File: chapter15/src/main/java/com/apress/prospring6/fifteen/controllers/RestErrorHandler.java
package com.apress.prospring6.fifteen.controllers;

import org.springframework.dao.DataIntegrityViolationException;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;

@RestControllerAdvice
public class RestErrorHandler {

    @ExceptionHandler(DataIntegrityViolationException.class)
    public ResponseEntity<HttpStatus> handleBadRequest(DataIntegrityViolationException ex) {
        return ResponseEntity.badRequest().build();
    }
}

File: chapter16-kotlin/src/main/kotlin/com/apress/prospring6/sixteen/kotlin/boot/RestErrorHandler.kt
package com.apress.prospring6.sixteen.kotlin.boot

import org.springframework.dao.DataIntegrityViolationException
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.ControllerAdvice
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.ResponseBody

@ControllerAdvice
@ResponseBody
class RestErrorHandler {

    @ExceptionHandler(DataIntegrityViolationException::class)
    fun handleBadRequest(ex: DataIntegrityViolationException): ResponseEntity<Any>? {
        return ResponseEntity.badRequest().body(ex.message)
    }
}